### PR TITLE
Doc: Datasets doc

### PR DESCRIPTION
**Pull request by codingl2k1**
## Related issue number

Fixes #xxxx

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

File: doc/source/libraries/xorbits_data/datasets.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    4     emerges as something rare , an issue movie tha...      1
+                                                    ...    ...
+    8525  any enjoyment will be hinge from a personal th...      0
+    8526  if legendary shlockmeister ed wood had ever ma...      0
+    8527  hardly a nuanced portrait of a young woman's b...      0
+    8528    interminably bleak , to say nothing of boring .      0
+    8529  things really get weird , though not particula...      0
+    [8530 rows x 2 columns]
+    >>> df["label"].sum()
+    4265
```
Comment (aresnow1) [inline doc/source/libraries/xorbits_data/datasets.rst:68]: Here can emphasize that operations on dataframe are also in parallel.

File: python/setup.cfg [hunk 2]
```diff
@@ -78,6 +79,7 @@ doc =
     xgboost>=1.3.0
     lightgbm>=3.3.5
     matplotlib
+    datasets
```
Comment (aresnow1) [inline python/setup.cfg:82]: Add a dataset extra requirements.

Comment (codingl2k1) [reply python/setup.cfg:82]: This is for sphinx to get correct source module. The `dataset` extra requirement is already there.

File: doc/source/libraries/xorbits_data/datasets.rst (hunk 1 above)
Comment (codingl2k1) [reply doc/source/libraries/xorbits_data/datasets.rst:86]: Good point. Thanks.

File: doc/source/libraries/xorbits_data/datasets.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+.. currentmodule:: xorbits.datasets
+
+This is a short introduction to :code:`xorbits.datasets`.
+
+Datasets Creation
+-----------------
+
+You can create a dataset from Hugging Face datasets, just like the `datasets.load_dataset`. But, 
+xorbits.datasets will load the dataset in parallel*. *(Currently, not all the datasets are loaded
```
[the remark this answers is not in the document]
Comment (codingl2k1) [reply doc/source/libraries/xorbits_data/datasets.rst]: Document updated. Thanks.

File: doc/source/libraries/xorbits_data/datasets.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+in parallel.)*
```
[the remark this answers is not in the document]
Comment (codingl2k1) [reply doc/source/libraries/xorbits_data/datasets.rst]: Document updated. Thanks.

File: doc/source/libraries/xorbits_data/datasets.rst [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    >>> dataset = xdatasets.from_huggingface("rotten_tomatoes", split="train")
+    >>> # Use rechunk() to improve the concurrency.
+    >>> dataset = dataset.rechunk(10)
+    >>> def add_prefix(example):
+    >>>     example["text"] = "Xorbits: " + example["text"]
+    >>>     return example
+    >>> # 10 processes applying `add_prefix` concurrently.
+    >>> dataset = dataset.map(add_prefix)
+    >>> # Currently, you have to execute() and fetch() to get all the dataset.
+    >>> dataset.execute()
```
Comment (qinxuye) [inline doc/source/libraries/xorbits_data/datasets.rst:51]: Is the `execute` required?

File: python/xorbits/datasets/dataset.py [hunk 6]
```diff
@@ -80,10 +82,62 @@ class Dataset(HasShapeTileable):
     type_name = "Dataset"
 
     def rechunk(self, num_chunks: int, **kwargs):
+        """Split the internal data into chunks.
+
+        Currently, rechunk is not fully supported.
```
Comment (qinxuye) [inline python/xorbits/datasets/dataset.py:87]: What does `fully supported` mean?

File: doc/source/libraries/xorbits_data/datasets.rst (hunk 5 above)
Comment (codingl2k1) [reply doc/source/libraries/xorbits_data/datasets.rst:51]: Currently, the dataset can be auto executed, so the `execute()` is not required. But the meta is not being updated correctly, so the `fetch()` is required.

I will remove this `execute()` from doc.

The next PR is almost ready, then `execute()` and `fetch()` are not required.

File: python/xorbits/datasets/dataset.py (hunk 6 above)
Comment (codingl2k1) [reply python/xorbits/datasets/dataset.py]: I have fixed the doc. Thanks.

File: doc/source/libraries/xorbits_data/datasets.rst [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    >>> dataset = dataset.map(add_prefix)
+    >>> # Currently, you have to fetch() to get the dataset info.
+    >>> dataset.fetch()
+    Dataset({
+        features: ['text', 'label'],
+        num_rows: 8530
+    })
+
+Datasets Outputs
+---------------
```
Comment (qinxuye) [inline doc/source/libraries/xorbits_data/datasets.rst:58]: This is not aligned.

Comment (codingl2k1) [reply doc/source/libraries/xorbits_data/datasets.rst]: Fixed. Thanks.

**Review by qinxuye — APPROVED** [review]
LGTM

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**